src/core/qgsmaplayer.h

```diff
@@ -73,13 +73,17 @@ class CORE_EXPORT QgsMapLayer : public QObject
      */
     QgsMapLayer( QgsMapLayer::LayerType type = VectorLayer, const QString &name = QString::null, const QString &source = QString::null );
 
-    virtual ~QgsMapLayer();
+    /** Copy constructor.
+     * @param mapLayer the map layer to copy
+     * @param deep to copy unique id if true
+     */
+    QgsMapLayer( const QgsMapLayer &mapLayer, bool deep = true );
```

Comment:
Because `QgsMapLayer` has some virtual pure methods and my compiler wont compile something which is creating an instance of an abstract class...

But I'm doubting myself so I'm gonna try...